syntax = "proto3";

package banking;


// The branch service definition.
service Branch {
  // delivers instructions to the branch
  rpc MsgDelivery (BranchRequest) returns (BranchReply) {}
}

// Branch request message
message BranchRequest {
  string type = 1;
  float balance = 2;
  string interface = 3;
  float money = 4;
  int32 id = 5;
  int32 clock = 6;
  int32 event_id = 7;
}

// Branch response message
message BranchReply {
  string type = 1;
  float balance = 2;
  string interface = 3;
  float money = 4;
  int32 id = 5;
  int32 clock = 6;
  int32 event_id = 7;
  string request_status = 8;
}
